test/framework/package.py

```diff
@@ -184,10 +184,16 @@ def test_active_pns(self):
 
         pns = ActivePNS()
 
-        # default: EasyBuild package naming scheme, pkg release 1
-        self.assertEqual(pns.name(ec), 'OpenMPI-2.1.2-GCC-6.4.0-2.28')
-        self.assertEqual(pns.version(ec), 'eb-%s' % EASYBUILD_VERSION)
-        self.assertEqual(pns.release(ec), '1')
+        if get_package_naming_scheme() == "EasyBuildPNS":
+            # default: EasyBuild package naming scheme, pkg release 1
+            self.assertEqual(pns.name(ec), 'OpenMPI-2.1.2-GCC-6.4.0-2.28')
+            self.assertEqual(pns.version(ec), 'eb-%s' % EASYBUILD_VERSION)
+            self.assertEqual(pns.release(ec), '1')
+        elif get_package_naming_scheme() == "EasyBuildDebFriendlyPNS":
```

Comment:
I was wondering how this test could be passing when there's a clear typo...and now I know. `EasyBuildDebFriendlyPNS` is available as a package MNS, but it is never actually set in the test suite, (so this code path is never followed)

Reply:
You can change the PNS by setting an environment variable, so that would mean adding a loop around the code inside the function:
```
    def test_active_pns(self):
        """Test use of ActivePNS."""
        for pns_type in ['EasyBuildDebFriendlyPNS', 'EasyBuildPNS']:
            os.environ['EASYBUILD_PACKAGE_NAMING_SCHEME'] = pns_type
            init_config(build_options={'silent': True})
            ....
            del os.environ['EASYBUILD_PACKAGE_NAMING_SCHEME']
```

Reply:
I've updated the previous comment to include the full code content

Reply:
It's the same as before but with a loop around the test function contents

Reply:
Implimented as suggested. Test passes for me..

Reply:
Should not 'os.environ['EASYBUILD_PACKAGE_NAMING_SCHEME']` really be `os.environ['TEST_EASYBUILD_PACKAGE_NAMING_SCHEME']`?

Reply:
I got that from the module naming scheme tests so I don't think so, I think it matters that you're setting from within the test suite. A good check would be to reintroduce the typo and see if the tests break

Reply:
I ran the debugger on it. It is executed.. That's why I know it works.